test/spmd/test_xla_sharding.py

```diff
@@ -452,6 +452,16 @@ def test_no_sharding(self):
     t3_expected = [9.0, 9.0, 9.0, 9.0, 9.0, 9.0, 9.0, 9.0]
     self.assertEqual(t3.tolist()[0], t3_expected)
 
+  @unittest.skipUnless(
+      xm.get_xla_supported_devices("TPU"),
+      f"Requires PJRT_DEVICE set to `TPU`.")
+  def test_hybrid_mesh(self):
```

Comment:
We probably can add test cases for MultiSlice but only enables them for the XL-ML tests?

Reply:
Is there a way to add XL ML tests here. I was planning to add them in the other repo.

Reply:
It depends on you. You can have a stond-alone test in other repo. Or you can have a test here which is skipped for single slice. Then you run the test in other repo through a config.

Reply:
@khatwanimohit let's add a unit test here, you can mock out `pjrt.global_device_attributes` to spoof the device attributes. It will be good to have some explicit unit tests to verify the logic, and we can have an E2E test in XL ML tests.